### PR TITLE
Implement estate catalogue importer, storage, and watcher

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Adds a full catalogue importer and reconciler, storage models, a git-based watcher, and an async workflow to materialise YAML catalogue data into estates, projects, components, component_edges, and repositories. It is atomic (one transaction per commit), idempotent per commit, and auditable.
- Introduces an asynchronous import workflow via a Dramatiq actor and a git-based watcher to enqueue imports when the catalogue HEAD changes.
- Provides storage models for catalogue entities and an init function to boot the storage schema.
- Adds tests and docs scaffolding for the importer, storage, and watcher.

## Changes
- New: ghillie/catalogue/importer.py
  - Implements CatalogueImporter with import_path and import_catalogue, Church-agnostic mapping from catalogue to estate/project/component/repository/edge tables, and CatalogueImportResult for auditing stats.
  - Ensures a single transaction per commit, idempotent upserts/prunes, and optional commit_sha audit via CatalogueImportRecord.
  - Exposes build_importer_from_url for building an importer from a database URL.
  - Exposes import_catalogue_job Dramatiq actor for asynchronous imports.
- New: ghillie/catalogue/storage.py
  - Defines portable SQLAlchemy ORM models: Estate, ProjectRecord, RepositoryRecord, ComponentRecord, ComponentEdgeRecord, CatalogueImportRecord.
  - Includes init_catalogue_storage(engine) to create tables if missing.
- New: ghillie/catalogue/watch.py
  - Adds GitCatalogueWatcher and CatalogueWatcherConfig to poll a git repo and enqueue imports on HEAD moves via import_catalogue_job.
- Updated: ghillie/catalogue/__init__.py
  - Exports CatalogueImporter, CatalogueImportResult, build_importer_from_url, import_catalogue_job, init_catalogue_storage, and storage models for easier reuse in applications and tests.
- Tests (new):
  - tests/unit/test_catalogue_importer.py: unit tests for importer behavior, idempotency, and error handling.
  - tests/unit/test_catalogue_watcher.py: unit tests for the git-based watcher enqueue behavior.
  - tests/features/catalogue_importer.feature and tests/features/steps/test_catalogue_importer_steps.py: BDD-style tests validating end-to-end importer behavior using a sample catalogue.
- Docs (content hints):
  - Prose in docs/ghillie-bronze-silver-architecture-design.md describing Phase 1.1.b catalogue importer and reconciler (estate-world) and its transaction/idempotency guarantees.

## Design & Behavior
- Atomicity: Each import run executes within a single database transaction; invalid catalogues fail fast and rollback entirely.
- Idempotency: Re-importing the same commit_sha is a no-op (skipped) and does not duplicate data.
- Upsert + prune: Projects/components/repos are upserted by key; missing entries are pruned; edges are rebuilt from the catalogue graph.
- Auditing: Processed commits are recorded in catalogue_imports for traceability.
- Async workflow: A Dramatiq actor (import_catalogue_job) executes imports; a GitCatalogueWatcher polls git rev-parse HEAD and enqueues new imports automatically.
- Storage layer: Introduces portable SQLAlchemy models (SQLite in tests, PostgreSQL in prod) for estates, projects, components, repositories, and edges.

## Testing Plan
- Unit tests cover importer correctness, idempotency, validation errors, and edge cases.
- Integration/BDD tests validate end-to-end scenario using the provided sample catalogue (examples/wildside-catalogue.yaml).
- Watcher tests verify commit detection and enqueue behavior with a stubbed broker.
- To run: pytest (and pytest-bdd for feature tests). Ensure dependencies in pyproject.toml are installed (sqlalchemy, aiosqlite, dramatiq).

## Migration & Compatibility
- This change adds a new catalogue subsystem and does not modify existing code paths outside the new module surface area.
- Requires database initialization via init_catalogue_storage(engine) before importer usage.

## Example Usage
- Build an importer from a database URL and import a catalogue path:
  - engine = create_async_engine(database_url, future=True)
  - await init_catalogue_storage(engine)
  - importer = CatalogueImporter(async_sessionmaker(engine, expire_on_commit=False), estate_key="default")
  - await importer.import_path(Path("path/to/catalogue.yaml"), commit_sha="abc123")
- Or enqueue via the Dramatiq actor import_catalogue_job for asynchronous processing.

## Notes
- Dependencies added to pyproject.toml include sqlalchemy>=2.0.34, aiosqlite>=0.20.0, and dramatiq>=1.16.0 to support storage, transactional import, and background processing.

🌿 Generated by [Terry](https://www.terragonlabs.com)

---
ℹ️ Tag @terragon-labs to ask questions and address PR feedback

📎 **Task**: https://www.terragonlabs.com/task/7122adba-89b6-40fa-b630-b898bdeceacd

### CODE REVIEW

File: ghillie/catalogue/importer.py [hunk 1]
```diff
@@ -0,0 +1,488 @@
...
+        async with self._session_factory() as session, session.begin():  # type: ignore[call-arg]
+            estate = await self._ensure_estate(session)
+            existing_import = None
+            if commit_sha:
+                existing_import = await session.scalar(
+                    select(CatalogueImportRecord).where(
+                        CatalogueImportRecord.commit_sha == commit_sha
+                    )
+                )
+                if existing_import:
```
[the remark this answers is not in the document]
Comment (leynos) [reply ghillie/catalogue/importer.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.

File: ghillie/catalogue/importer.py [hunk 2]
```diff
@@ -0,0 +1,663 @@
...
+                        description=component.description,
+                        notes=list(component.notes),
+                    )
+                    session.add(comp_record)
+                    result.components_created += 1
+                    existing_components[component.key] = comp_record
+
+                component_index[component.key] = comp_record
+
+            for key, comp_record in list(existing_components.items()):
```
[the remark this answers is not in the document]
Comment (leynos) [reply ghillie/catalogue/importer.py]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API.
